pkg/appgw/internaltypes.go

```diff
@@ -65,6 +65,7 @@ type serviceBackendPortPair struct {
 type listenerIdentifier struct {
 	FrontendPort int32
 	HostName     string
+	UsePrivateIP bool
```

Comment:
Now that this 3rd items is part of the identity struct - should we include it in the listener name too?
It'd be good to augment the listener name from `fl-bye.com-80` to something like `fl-bye.com-80-privip` or `fl-bye.com-80-publip`